### DISCUSSION TITLE
Can I create multiple python versions in airflow container

### DISCUSSION THREAD

**Discussion by yeangPan** [apache/airflow#34961] [Q&A] [answered]
the airflow need to run many tasks, whose python version was different from each other, How can I create different python version in Dockerfile for the container when I deployed my airflow using docker compose ?

**Comment by potiuk** [comment, marked as answer]
Just like in any other container. You need to build your own image and install python and create venvs in it. Nothing specific to Airflow, you are likely to find many guides for that. but you might follow some of the ideas and examples on extending image from https://airflow.apache.org/docs/docker-stack/build.html#extending-the-image - there is no specific "add your own python" case but if you find out how, contributing an example how to do it it to the docs might be a good idea.

**Reply by yeangPan** [reply]
after reading the docs, it finally worked, many thx!
